docs/observability/index.mdx

```diff
@@ -0,0 +1,301 @@
+
+<CodeTabs
+  groupId="client-language"
+  tabs={[
+    python({ label: "Python" })`
+      from openai import OpenAI
+      # highlight-next-line
+      from langsmith import traceable
+      from langsmith.wrappers import wrap_openai
+      
+      openai_client = wrap_openai(OpenAI())
+      
+      def retriever(query: str):
+          results = ["Harrison worked at Kensho"]
+          return results
+      
+      # highlight-next-line
+      @traceable
+      def rag(question):
+          docs = retriever(question)
+          system_message = """Answer the users question using only the provided information below:
+          
+          {docs}""".format(docs="\\n".join(docs))
+          
+          return openai_client.chat.completions.create(
+              messages=[
+                  {"role": "system", "content": system_message},
+                  {"role": "user", "content": question},
+              ],
+              model="gpt-4o-mini",
+          )
+    `,
+    typescript({ label: "TypeScript" })`
+      import { OpenAI } from "openai";
+      // highlight-next-line
+      import { traceable } from "langsmith/traceable";
+      import { wrapOpenAI } from "langsmith/wrappers";
+      
+      const openAIClient = wrapOpenAI(new OpenAI());
+      
+      async function retriever(query: string) {
+        return ["This is a document"];
+      }
+      
+      // highlight-next-line
+      const rag = traceable(async function rag(question: string) {
+        const docs = await retriever(question);
+        
+        const systemMessage =
+          "Answer the users question using only the provided information below:\\n\\n" +
+          docs.join("\\n");
+          
+        return await openAIClient.chat.completions.create({
+          messages: [
+            { role: "system", content: systemMessage },
+            { role: "user", content: question },
+          ],
+          model: "gpt-4o-mini",
+        });
+      });
+    `,
+  ]}
+/>
+
+Now if you call your application as follows:
+
+```python
+rag("where did harrison work")
+```
+
+This will produce a trace of just the entire pipeline (with the OpenAI call as a child run) - it should look something like [this](https://smith.langchain.com/public/2174f4e9-48ab-4f9e-a8c4-470372d976f1/r)
+
+![](./tutorials/static/tracing_tutorial_chain.png)
+
+## Next steps
+
+Congratulations! If you've made it this far, you're well on your way to being an expert in observability with LangSmith.
+Here are some topics you might want to explore next:
+
+- [Trace a `LangChain` application](./observability/how_to_guides/tracing/trace_with_langchain)
+- [Trace a `LangGraph` application](./observability/how_to_guides/tracing/trace_with_langgraph)
+- [Trace multiturn conversations](./observability/how_to_guides/monitoring/threads)
+- [Send traces to a specific project](./observability/how_to_guides/tracing/log_traces_to_project)
```

Comment:
/observability/how_to_guides/monitoring/filter_traces_in_application

this could be a good one for next steps too